rmf_site_format/src/mobility.rs

```diff
@@ -16,32 +16,72 @@
 */
 
 #[cfg(feature = "bevy")]
-use bevy::prelude::{Component, Reflect, ReflectComponent};
+use bevy::prelude::{Component, Reflect};
 use serde::{Deserialize, Serialize};
 
-#[derive(Serialize, Deserialize, Debug, Clone, Copy, Default, PartialEq, Eq)]
-#[cfg_attr(feature = "bevy", derive(Component, Reflect))]
-#[cfg_attr(feature = "bevy", reflect(Component))]
-pub struct MobileRobotMarker;
+#[derive(Serialize, Deserialize, Debug, Clone, PartialEq)]
+#[cfg_attr(feature = "bevy", derive(Component))]
+pub struct Mobility {
+    pub kind: String,
+    pub config: serde_json::Value,
+    pub bidirectional: bool,
+    pub collision_radius: f32,
+    pub rotation_center_offset: [f32; 2],
```

Comment:
I think we should put all these fields (`bidirectional`, `collision_radius`, and `rotation_center_offset`) into `struct DifferentialDrive` since they might not make sense for other forms of mobility, e.g. omni wheel drive.

I'm not certain that `collision_radius` belongs inside of `Mobility` at all.. collision information probably deserves its own component. But let's keep it in `DifferentialDrive` for now until we can think it over more.

Reply:
With reference to the comment above, if we were to simplify `RobotMarker`/`OptionalModelData` to `Robot`, `collision_radius` might go well as a field under `Robot` alongside `skills`.

Reply:
Yeah I was thinking about what it should look like as its own thing parallel to `Mobility`.

Here's one way I can imagine it working:
* Instead of `Robot::skills` we can call it `Robot::properties`
* We introduce a `Collision`  struct similar to `Mobility` that has a `kind` and `config`
* We have a circle collision kind:

```rust
pub struct CircleCollision {
    pub radius: f32,
    pub offset: Vec2,
}
```